SPEC.md

```diff
@@ -314,6 +324,36 @@ String is either a [Quoted String](#quoted-string) or a
 [Raw String](#raw-string). Quoted Strings may include escaped characters, while
 Raw Strings always contain only the literal characters that are present.
 
+Strings _MUST_ be represented as UTF-8 values.
+
+Strings _MUST NOT_ include the code points for [disallowed literal
+code points](#disallowed-literal-code-points) directly. If needed, they can be
+specified with their corresponding `\u{}` escape.
+
+### Multi-line Strings
+
+Strings may span multiple lines with literal Newlines, in which case the
+resulting String is "dedented" according to the line with the fewest number of
+Whitespace characters preceding the first non-Whitespace character. That is,
+the number of Whitespace characters in the least-indented line in the String
+body is subtracted from the Whitespace of all other lines.
+
+Multi-line strings _MUST_ have a single [Newline](#newline) immediately
+following their opening `"`, after which they may have any number of newlines.
+Finally, there must be a Newline, followed by any number of Whitespace, before
+the closing `"`.
+
+The first Newline, the last Newline, along with Whitespace following the last
+Newline, are not included in the value of the String. The first and last
+Newline can be the same character (that is, empty multi-line strings are
+legal).
```

Comment:
Just checking: the following is *also* an empty string, right?

```
node "

"
```

(that is, `"\n\n"`)

It *should* be, fwiw, just verifying.

Reply:
See our discussion above about these cases

Reply:
hmmmm. You're right. Based on what I wrote, this would be an empty string, but I would expect that node to be equivalent to `node "\n"`.

Reply:
Oh, actually, the JS spec implementation also treats that as `""`, not `"\n"`. Interesting.

Reply:
Right, the following

```
node "
	foo
	"
```

should be equivalent to `node "foo"`, so my example should similarly just be the empty string. I was just double-checking and wondering if we wanted to make that a little clearer.